d2bs/kolbot/libs/common/Attacks/Bone.js

```diff
@@ -0,0 +1,235 @@
+/**
+ *	@filename	Bone.js
+ *	@author		jaenster
+ *	@desc		A optimized bone spear attack sequence
+ */
+
+var ClassAttack = (function () {
+	// Load GameData
+	include('GameData.js');
+	const Result = {
+		FAIL: 0,
+		OK: 1,
+		NO_VALID_SKILLS: 2,
+		NEXT_TARGET: 3,
+	};
+
+	// I dont like magic numbers
+	const sdk = require('../../modules/sdk');
+
+	// Might add some generics in the future
+	const Generics = {};
+
+	const Bone = Object.create(Generics);
+
+	// What we need to check?
+	const commonChecks = () => {
+		// 1) Our bone armor still up?
+		if (me.getSkill(sdk.skills.BoneArmor, 1) && !me.getState(sdk.states.BoneArmor)) {
+			Skill.cast(sdk.skills.BoneArmor);
+		}
+		// 2) Our merc still alive?
+		if (Config.MercWatch && Town.needMerc()) {
+			let [act, area] = [me.act, me.area];
+
+			// I dont want to visit town and pot up and whatever, i just wanna revive merc and go
+			Town.goToTown() && Town.reviveMerc() && Town.goToTown(act) && Town.move("portalspot");
+
+			const portal = Pather.getPortal(area, me.name);
+			if (!portal) throw new Error('Failed to go back after reviving merc');
+			Pather.usePortal(undefined, undefined, portal);
+		}
+		// 3) Need to dim someone's vision?
+		//ToDo actually write this properly
+
+		// 4) who is our merc attacking?
+		let merc = me.getMerc();
+		if (merc) {
+			const unit = getUnits(sdk.unittype.Monsters)
+				.filter(unit => {
+					if (!Attack.checkMonster(unit)) {
```

Comment:
combine lines 50-62 with 
condition || condition || ..

would change fat arrow to regular function syntax as well